### DISCUSSION TITLE
why does __strimzi-topic-operator-kstreams-topic-store-changelog  has Replication factor=1

### DISCUSSION THREAD

**Discussion by dbaarorapr** [strimzi/strimzi-kafka-operator#6908] [General]
why does __strimzi-topic-operator-kstreams-topic-store-changelog  has Replication factor=1

How shall this Topic have a High Availability with  ReplicationFactor: 1	

And when we do rolling restarts -- we do get offline partitions alerts and cannot find an associated Topic name

We are on Strimzi versions 0.26.1 and trying to understand the reason why this has ReplicationFactor: 1	?

```
bin/kafka-topics.sh --zookeeper localhost:12181 --describe --topic __strimzi-topic-operator-kstreams-topic-store-changelog
Topic: __strimzi-topic-operator-kstreams-topic-store-changelog	TopicId: JhiDaYJeSueTT5KBwD1Jwg	PartitionCount: 1	ReplicationFactor: 1	Configs: cleanup.policy=compact,min.compaction.lag.ms=0,message.timestamp.type=CreateTime,segment.bytes=67108864
	Topic: __strimzi-topic-operator-kstreams-topic-store-changelog	Partition: 0	Leader: 1	Replicas: 1	Isr: 1

```

**Comment by scholzj** [comment]
It is created by Kafka Streams. I think it is using the default replication settings of your Kafka cluster. Or @tomncooper and @LiamClarkeNZ?

**Comment by dbaarorapr** [comment]
so in this cluster i  checked and we have 
default.replication.factor = 3

as seen in this startup logs


k logs shared-kafka-kafka-0   |grep default.replication.factor
default.replication.factor=3
	default.replication.factor = 3
	default.replication.factor = 3

**Comment by dbaarorapr** [comment]
also kafka CR object has below config

    config:
      auto.create.topics.enable: false
      default.replication.factor: 3
      num.partitions: 3
      offsets.topic.replication.factor: 3
      transaction.state.log.min.isr: 2
      transaction.state.log.replication.factor: 3

**Comment by LiamClarkeNZ** [comment]
Hi @dbaarorapr unfortunately, the topic operator version you're using relies on the Kafka Streams defaults for replication factor for its internal topics - which in earlier versions of Kafka Streams was 1. Later versions of the topic operator use a version of Kafka Streams that correctly defaults to the default replication factor of the cluster. 

A workaround for this issue is to increase replication factor for those topics using kafka-reassign-partitions.

**Reply by dbaarorapr** [reply]
Thanks for getting back quickly

Can you please confirm what version of strimzi is having the  change  that uses default RF?

Also,if we upgrade ,shall this get fixed or we still need to increase RF with kafka-reassign-partitions.sh

Regards
Praveen

**Reply by scholzj** [reply]
@LiamClarkeNZ should confirm. But I don't think upgrade wil reconfigure the topic. You would need to do it manually or delete it after the upgrade to have it recreated.

**Reply by dbaarorapr** [reply]
Thanks 

So do we have have any data loss or issues ,if we delete it after upgrade

shall it get created automatically after deletion without any effects  to the strimzi kafka cluster?
